src/containers/OrderSelection.tsx

```diff
@@ -0,0 +1,189 @@
+  saveFittingHeight: typeof orderActions.saveFittingHeightStart;
+};
+
+type ComponentProps = PropsFromState & PropsFromDispatch;
+
+const mapStateToProps = (state: ApplicationState) => ({
+  isFetching: state.order.isFetching || state.lenses.isFetching || state.frames.isFetching,
+  state: state,
+  errors: state.order.errors,
+  frameErrors: state.frames.errors,
+  order: state.order,
+  selectedLens: state.order.boss.lens,
+  prescription: state.order.boss.prescription,
+});
+
+const mapDispatchToProps = (dispatch: Dispatch) => ({
+  handleOpen: () => dispatch(framesActions.open()),
+  submitOrder: (boss: Boss) => dispatch(orderActions.submitStart(boss)),
+  saveOrder: (state: ApplicationState) => dispatch(orderActions.saveOrderStart(state)),
+  saveFittingHeight: (boss: Boss, height: number) => dispatch(orderActions.saveFittingHeightStart(boss, height)),
+});
+
+class OrderSelection extends LinkComponent<ComponentProps> {
+  public componentDidUpdate(prevProps: ComponentProps) {
+    const { order } = this.props;
+
+    if (order.redirect) {
+      this.redirectToPage('/order');
+    }
+  }
+
+  public render() {
+    const {
+      isFetching,
+      errors,
+      frameErrors,
+      state,
+      order,
+      handleOpen,
+      submitOrder,
+      saveOrder,
+      saveFittingHeight
+    } = this.props;
+    const frameSelectionButtonDisabled = isEmptyObject(order.boss.lens);
+    const isFrameSelected = !isEmptyObject(order.boss.frame);
+    const isFittingHeightSelected = order.boss.fittingHeight ? true : false;
+    const isErrors = frameErrors ? true: false;
+    const submitDisabled =  !isFittingHeightSelected || isErrors;
+
+    return (
+      <main className="p-template__main">
+      {isFetching && <CircularProgress className="p-template__progress"/>}
+        <PrescriptionSelectionContainer />
+
+        {isFrameSelected
+          ? (
+            <React.Fragment>
+              <Section tittle="Enter fitting height" wrap>
+                {errors.fittingHeight && <div className="order-selection__error">{errors.fittingHeight}</div>}
+                <SelectField
+                  disabled={isFetching}
+                  className="order-selection__select"
+                  label="Height"
+                  value={order.boss.fittingHeight}
+                  list={order.fittingProperties}
+                  onChange={(value: number) => saveFittingHeight(order.boss, value)}
+                />
+              </Section>
+              <Section className="order-selection__info s-template__content" tittle="Message">
+                <p>{order.message}</p>
+              </Section>
+            </React.Fragment>
+          )
+          : (
+            <Section className="order-selection__info s-template__content" tittle="Recommendation">
+              <p>{order.recommendation}</p>
+            </Section>
+          )
+        }
+
+        <LensSelectionContainer />
+        <FrameSelection />
+
+        {isFrameSelected
```

Comment:
split to `renderSelected` and `renderUnselected`, but you defenetly know more understandable method names